src/coreclr/vm/prestub.cpp

```diff
@@ -1004,6 +1004,11 @@ PCODE MethodDesc::JitCompileCodeLocked(PrepareCodeConfig* pConfig, COR_ILMETHOD_
         InterpreterPrecode* pPrecode = InterpreterPrecode::FromEntryPoint(pCode);
         InterpByteCodeStart* interpreterCode = dac_cast<InterpByteCodeStart*>(pPrecode->GetData()->ByteCodeAddr);
         pConfig->GetMethodDesc()->SetInterpreterCode(interpreterCode);
+
+        // Ensure the method has a call stub. We have to do this eagerly here.
```

Comment:
It was here originally and then changed to be lazier based on feedback. Majority of methods won't ever need these stubs, so it is wasteful to create them eagerly.

Reply:
My attempt before this did it lazily inside the interpreter stub itself, but that was complex to get right. Do you think we should do it lazily in the interpreter stub?

Reply:
It would be useful to understand what leads to the crash in detail. Given that the crash is non-deterministic, we may have a race condition somewhere. I would hope that fixing the race condition should not require a pile of new assembly code to be written.

Both this commit and the previous commit look like workarounds for the actual bug.

Reply:
For me it failed 100% of the time when debugging and was successful 100% of the time without a debugger attached. I agree that it does seem like it would have to be a race condition, but the test case I'm running doesn't use threads. Maybe it's tiered compilation?

Reply:
If you are running under VS debugger, VS debugger tends to enable extra tracing that may explain what you are seeing.